src/components/CountryFilter.jsx

```diff
@@ -0,0 +1,22 @@
+// IMPORTS
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchCountries } from '../redux/countrySlice';
+
+// COMPONENTS
+const CountryFilter = () => {
+  const dispatch = useDispatch();
+
+  // SELECT STATE FROM REDUX STORE
+  const countries = useSelector((state) => state.countries.countries);
+  // USE USEEFEECT TO DISPATCH FETCHCOUNTRIES ACTION
+  useEffect(() => {
+    dispatch(fetchCountries());
+  }, [dispatch]);
+
+  // FILTER COUNTRIES BY REGION
+  const filteredCountries = countries.filter((country) => country.region === 'Asia');
```

Comment:
- [ ] Hello, as mentioned by the previous reviewer and a requirement in the project guideline,To enhance user experience, let's incorporate a search field on the homepage, allowing users to filter the displayed list of countries based on their input. Simply add a search input and implement state management for smooth list filtering. This addition ensures users can easily explore and find specific countries, enhancing the overall usability of the platform. 🕵️‍♂️🔍 Happy coding! 💻🚀

![Screenshot from 2023-11-17 20-24-54](https://github.com/HossainAraf/metrics-webapp/assets/112261237/6e3f6c78-135d-4d8a-b938-c269edff9003)